### PR TITLE
Add spring cloud config name property

### PR DESCRIPTION
**Pull request by johnowl**
This is a possible solution for the issue #230 that I opened.

### CODE REVIEW

File: discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudConfigurationClient.java [hunk 1]
```diff
@@ -89,7 +89,8 @@ protected SpringCloudConfigurationClient(SpringCloudConfigClient springCloudConf
         if (!configuredApplicationName.isPresent()) {
             return Flowable.empty();
         } else {
-            String applicationName = configuredApplicationName.get();
+            String applicationName = springCloudConfiguration.getName() == null ?
+                    configuredApplicationName.get() : springCloudConfiguration.getName();
```
Comment (jeffbrown) [inline discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudConfigurationClient.java:93]: This is a minor thing but `springCloudConfiguration.getName()` could be extracted to a local variable so you won't have to invoke that method twice.

Comment (johnowl) [reply discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudConfigurationClient.java:93]: Ok, I am going to refactor it.

Comment (johnowl) [reply discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudConfigurationClient.java]: @jeffbrown thanks for your review, the code was updated.

File: discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudClientConfiguration.java [hunk 2]
```diff
@@ -105,6 +106,13 @@ public String getLabel() {
         return label;
     }
 
+    /**
+     * @return The spring cloud config server name
+     */
+    public String getName() {
```
Comment (jameskleeh) [inline discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudClientConfiguration.java:112]: This should return an optional

**Comment by jameskleeh** [comment]
Needs docs

File: discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudClientConfiguration.java (hunk 2 above)
Comment (johnowl) [reply discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudClientConfiguration.java]: done

**Comment by johnowl** [comment]
> Needs docs

Docs added

**Pull request merged**